### ISSUE TITLE
Problems calling async_forward_entry_setup

### ISSUE THREAD

**Issue by jkfranks9** [mawinkler/astroweather#59]
Receiving 2 seemingly related error messages at startup:

```
2024-07-04 09:56:07.060 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'astroweather' calls async_forward_entry_setup for integration, astroweather with title: [NETWORK] and entry_id: 5eaaf04d1261eb307eb7aa44c1c312cb, which is deprecated and will stop working in Home Assistant 2025.6, await async_forward_entry_setups instead at custom_components/astroweather/__init__.py, line 263: hass.async_create_task(, please create a bug report at https://github.com/mawinkler/astroweather/issues
2024-07-04 09:56:07.121 WARNING (MainThread) [homeassistant.helpers.frame] Detected code that calls async_forward_entry_setup for integration astroweather with title: [NETWORK] and entry_id: 5eaaf04d1261eb307eb7aa44c1c312cb, during setup without awaiting async_forward_entry_setup, which can cause the setup lock to be released before the setup is done. This will stop working in Home Assistant 2025.1. Please report this issue.
Stack (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2122, in async_forward_entry_setup
    _report_non_awaited_platform_forwards(entry, "async_forward_entry_setup")
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1175, in _report_non_awaited_platform_forwards
    report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 151, in report
    _LOGGER.warning(msg, stack_info=True)
```

**Comment by mawinkler** [comment]
Will take care on this

**Comment by heinemannj** [comment]
Above warnings also reported on my setup - I'm available for beta testing ...

**Comment by mawinkler** [comment]
No new release yet, will likely do later today. Issue should be fixed in main.

**Comment by mawinkler** [comment]
Should be fixed with [0.50.4](https://github.com/mawinkler/astroweather/releases/tag/v0.50.4).

**Comment by jkfranks9** [comment]
Fix confirmed, thanks!
